src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number
+}
 
+function checkData(...arr: number[]): void {
+  if (arr.some((el) => el <= 0)) {
+    throw new Error('Data can not be less then zero');
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkData(a, b, c);
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('Wrong data:'
+      + 'You can not build a triangle with these sides');
+    }
+  }
 
+  shape = Shape.Triangle;
```

Comment:
```suggestion
  readonly shape = Shape.Triangle;
```
You can make this prop protected.